qml/WordRibbon.qml

```diff
@@ -36,51 +37,73 @@ Rectangle {
-
-            Item {
-                anchors.fill: parent
-                anchors.margins: {
-                    top: 0
-                    bottom: 0
-                    left: Device.gu(2)
-                    right: Device.gu(2)
-                }
+            spacing: listView.spacing
+
+            Label {
+                id: wordItem
+
+                Layout.alignment: Qt.AlignVCenter | Qt.AlignHCenter
+                Layout.fillHeight: true
+
+                property bool textBold: isPrimaryCandidate || listView.count == 1 // Exposed for autopilot
+
+                horizontalAlignment: Text.AlignHCenter
+                verticalAlignment: Text.AlignVCenter
 
-                Label {
-                    id: wordItem
-                    font.pixelSize: Device.wordRibbonFontSize
-                    font.family: Theme.fontFamily
-                    font.weight: textBold ? Font.Bold : Font.Light
-                    text: word;
-                    anchors.centerIn: parent
-                    color: Theme.fontColor
+                font.pixelSize: height - Device.top_margin * 2
+                font.family: Theme.fontFamily
+                font.weight: textBold ? Font.Bold : Font.Light
+                text: word
+                color: Theme.fontColor
+
+                MouseArea {
+                    anchors.fill: parent
+                    onPressed: {
+                        Feedback.keyPressed();
+
+                        wordRibbonCanvas.state = "SELECTED"
+                        event_handler.onWordCandidatePressed(wordItem.text, isUserInput)
+                    }
+                    onReleased: {
+                        wordRibbonCanvas.state = "NORMAL"
+                        event_handler.onWordCandidateReleased(wordItem.text, isUserInput)
+                    }
                 }
             }
 
-            MouseArea {
-                anchors.fill: wordCandidateItem
-                onPressed: {
-                    Feedback.keyPressed();
-                    
-                    wordRibbonCanvas.state = "SELECTED"
-                    event_handler.onWordCandidatePressed(wordItem.text, isUserInput)
-                }
-                onReleased: {
-                    wordRibbonCanvas.state = "NORMAL"
-                    event_handler.onWordCandidateReleased(wordItem.text, isUserInput)
+            Component.onCompleted: {
+                // Use 1/3 of pixel height of parent converted to grid units
+                // as a minimum width threshhold, so that short suggestions
+                // are wide enough to tap with a thumb
+                let minWidth = Device.gu(parent.height / 3);
+                if (wordCandidateItem.width < minWidth) {
+                    wordCandidateItem.width = minWidth;
                 }
             }
+
+            // FIXME: ListView doesn't support separators directly so we need
```

Comment:
I think you could actually use ListView sections for this, if you set `section.property: index` and use a rectangle as the `section.delegate`. Not sure if it's worth the trouble though.

Reply:
I think it probably won't be very reliable (there are some cases where I've seen the same suggestion come up twice, which would lead to some weirdness using sections where the section name is the suggested completion).

Reply:
Right, yeah, probably not worth the trouble then.